src/NuGet.Status/Helpers/StorageService.cs

```diff
@@ -1,52 +1,103 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved. 
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information. 
 
-using Microsoft.WindowsAzure.Storage;
-using Microsoft.WindowsAzure.Storage.Blob;
-using Microsoft.WindowsAzure.Storage.Table;
 using System;
+using Azure.Data.Tables;
+using Azure.Identity;
+using Azure.Storage.Blobs;
 
 namespace NuGet.Status.Helpers
 {
     public class StorageService
     {
-        private readonly Func<string> _getConnectionString;
+        private readonly bool _useManagedIdentity;
+        private readonly string _managedIdentityClientId;
+        private readonly Func<string> _getBlobConnectionString;
+        private readonly Func<string> _getTableConnectionString;
 
         public string Name { get; }
 
-        public StorageService(string name, Func<string> getConnectionString)
+        // Do we need to use lazy loading for connection strings? We're not using KV secret injection for SAS tokens any more, does that change things?
+        // Do we need to pass config values in the ctor at all? Looks like we can just access the App config values directly in the methods (MvcApplication.Configuration..).
+        public StorageService(
+            string name,
+            string useManagedIdentityStr,
+            string managedIdentityClientId,
+            Func<string> getBlobConnectionString,
+            Func<string> getTableConnectionString)
         {
             Name = name;
-            _getConnectionString = getConnectionString;
+            _useManagedIdentity = false;
+            if (!bool.TryParse(useManagedIdentityStr, out _useManagedIdentity))
+            {
+                throw new ArgumentException("Invalid value for Storage:UseManagedIdentity. Expected 'true' or 'false'.", nameof(useManagedIdentityStr));
+            }
+
+            if (!string.IsNullOrWhiteSpace(managedIdentityClientId))
+            {
+                _managedIdentityClientId = managedIdentityClientId;
+            }
+
+            _getBlobConnectionString = getBlobConnectionString;
+            _getTableConnectionString = getTableConnectionString;
         }
 
-        public CloudBlockBlob GetCloudBlockBlob()
+        public BlobClient GetBlobClient()
         {
-            var container = GetCloudBlobContainer();
-            return container.GetBlockBlobReference(MvcApplication.StatusConfiguration.BlobName);
+            BlobContainerClient containerClient = GetBlobContainerClient();
```

Comment:
Out of curiosity, is there a reason that we explicitly recreate these clients (BlobContainerClient and BlobServiceClient) on every call of this method?

I think this was potentially necessary in the past to avoid re-using an expired SAS token, but with MI, this shouldn't be an issue anymore.